Python/flowgraph.c

```diff
@@ -1586,6 +1586,54 @@ optimize_cfg(cfg_builder *g, PyObject *consts, PyObject *const_cache)
     return SUCCESS;
 }
 
+static void
+make_super_instruction(cfg_instr *inst1, cfg_instr *inst2, int super_op)
+{
+    /* Skip if instructions are on different lines */
+    if (inst1->i_loc.lineno != inst2->i_loc.lineno) {
+        return;
+    }
+    if (inst1->i_oparg >= 16 || inst2->i_oparg >= 16) {
+        return;
+    }
+    INSTR_SET_OP1(inst1, super_op, (inst1->i_oparg << 4) | inst2->i_oparg);
+    INSTR_SET_OP0(inst2, NOP);
+}
+
+static void
+insert_superinstructions(cfg_builder *g)
```

Comment:
Regarding terminology... maybe "combined instructions"?

```suggestion
insert_combined_instructions(cfg_builder *g)
```

Reply:
I think these are still "superinstructions" in the normal usage of the term, even if they are inserted by the compiler instead of by the specializer. I'd be in favor of continuing to use the term, but getting out of the mental habit of assuming "superinstruction" == "specializer-inserted superinstruction."

Reply:
(Nowadays it's more like `code-object-creation-inserted-superinstruction`.)